src/xcode/ENA/ENA/Source/Models/Task Scheduling/ENATaskScheduler.swift

```diff
@@ -48,6 +48,11 @@ final class ENATaskScheduler {
 	// MARK: - Task registration.
 	@available(iOS 13.0, *)
 	private func registerTask(with taskIdentifier: ENATaskIdentifier, execute: @escaping ((BGTask) -> Void)) {
+		guard !CWAHibernationProvider.shared.isHibernationState else {
```

Comment:
Instead of having this logic in the function doesn’t it make sense that we check the state before calling the function? It will be more cleaner I believe.

Reply:
As long as the function is `private` -- yes, I would agree to your approach. 👍